Mrc-4908 k8s setup (#1)
reside-ic/shiny-dev: README.md

This is a temporary repo designed to give an easy-to-understand deployment of the apache / haproxy / shiny stack; it does not include configurable applications or anything like that. See it online at https://shiny-dev.dide.ic.ac.uk (DIDE network only).

## Running in Kubernetes

The following is guide to run the shiny server in kubernetes. They are based on running a Kind cluster. The configuration may

> nee to be adjusted for other k8s clusters.

Reviewer: ```suggestion
need to be adjusted for other k8s clusters.
```

### Prerequisites

> A production kubernetes cluster using k3s is needed to be setup first. To setup a k8s cluster follow the guide [here](https://mrc-ide.myjetbrains.com/youtrack/articles/RESIDE-A-31/Setting-up-Kubernetes-k8s-Cluster). Note: If using dev cluster (KIND), the storageClassName: local-path & longhorn is unavailable and you will need to provision own storage class and persistant volume. 

Reviewer: ```suggestion
A production kubernetes cluster using k8s is needed to be setup first. To setup a k8s cluster follow the guide [here](https://mrc-ide.myjetbrains.com/youtrack/articles/RESIDE-A-31/Setting-up-Kubernetes-k8s-Cluster). Note: If using dev cluster (KIND), the storageClassName: local-path & longhorn is unavailable and you will need to provision own storage class and persistent volume. 
```

> A production kubernetes cluster using k3s is needed to be setup first. To setup a k8s cluster follow the guide [here](https://mrc-ide.myjetbrains.com/youtrack/articles/RESIDE-A-31/Setting-up-Kubernetes-k8s-Cluster). Note: If using dev cluster (KIND), the storageClassName: local-path & longhorn is unavailable and you will need to provision own storage class and persistant volume. 

Reviewer: > storageClassName: local-path & longhorn is unavailable and you will need to provision own storage class and persistant volume.

Sorry, don't really understand this. Why aren't these available? You talk about using `local-path` below, in persistance.yaml.

Author: have updated to avoid confusion

> Run `start-k8s-shiny <env>` to run the shint server in k8s.

Reviewer: ```suggestion
Run `./start-k8s-shiny <env>` to run the shiny server in k8s.
```

## Bringing the bits up

> ```bash

Reviewer 2: Suggest you remove the docker instructions and the haproxy/httpd bits entirely

Teardown

```bash
docker rm -f haproxy apache shiny-1
docker network rm twinkle
```

## Running in Kubernetes

The following is guide to run the shiny server in kubernetes. They are based on running a Kind cluster. The configuration may
nee to be adjusted for other k8s clusters.

### Prerequisites

> A production kubernetes cluster using k3s is needed to be setup first. To setup a k8s cluster follow the guide [here](https://mrc-ide.myjetbrains.com/youtrack/articles/RESIDE-A-31/Setting-up-Kubernetes-k8s-Cluster). Note: If using dev cluster (KIND), the storageClassName: local-path & longhorn is unavailable and you will need to provision own storage class and persistant volume. 

Reviewer 2: What does the recommended setup look like for developing this locally? or on a staging machine (separate from the shiny staging, which is really for the users)? Do we need a second cluster somewhere that we use?

Author: the link shows the dev and prod setup and they both can be used locally... for staging, we can use the same cluster and just do another deployment and rename the deployment to something else... But if you wanted a seperate machine for just staging spinning up another cluster might be easiest
